plugin/edge/services.go

```diff
@@ -0,0 +1,53 @@
+package edge
+
+import (
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Starts the process of reading Kubernetes services using a list watcher.
+func (e *Edge) startReadingServices() {
+	go func() {
+		w, err := e.clientset.CoreV1().Services("").Watch(metaV1.ListOptions{Watch: true})
+		if err != nil {
+			log.Fatalf("couldn't read locally running Kubernetes services: %v", err)
+		}
+		e.watcher = w
+		for {
+			eventChan := e.watcher.ResultChan()
+			for rawEvent := range eventChan {
+
+				// Convert the watch event to our own service event type.
+				event, err := parseEvent(rawEvent)
+				if err != nil {
+					log.Errorf("couldn't read locally running Kubernetes services: %v", err)
```

Comment:
the append string "couldn't read..." on this specific parse error is the same as line 12 ^^, nit, maybe make the error message more specific